lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -193,10 +193,11 @@ public APIGatewayProxyResponseEvent handleRequest(
                     ipvSessionService.generateIpvSession(
                             clientOAuthSessionId, null, emailAddress, isReverification);
 
+            Optional<JarUserInfo> jarUserInfoClaim = getJarUserInfo(claimsSet);
             String evcsAccessToken = null;
             if ((configService.enabled(EVCS_READ_ENABLED)
-                    || configService.enabled(EVCS_WRITE_ENABLED))) {
-                Optional<JarUserInfo> jarUserInfoClaim = getJarUserInfo(claimsSet);
+                            || configService.enabled(EVCS_WRITE_ENABLED))
+                    && (jarUserInfoClaim.isPresent())) {
```

Comment:
Do you need this additional check? If anything I think it's problematic because it means we'll silently ignore a missing access token if we need it.

Reply:
yeah don't need this check.....as it'll already be taken care in validation.